File: python/bc2523.py
while True:
    try:
        cLe = input().strip()
        num = int(input().strip())

        cVal = [0]*num

        cPos = input().strip().split(" ")

        for c in range(0, num):
            cVal[c] = int(cPos[c])-1

        res = ""

        for d in range(0, num):
            pos = cVal[d]
            res += cLe[pos]

        print(res)

    except EOFError:
        break

File: cpp/bc2523.cpp
#include <iostream>
#include <vector>
#include <sstream>
#include <cmath>

using namespace std;

int main()
{
	string cLe, cPos, token, res;
	vector<int> vec;
	int num, pos;
	
	while(cin >> cLe)
	{
	    res = "";
		cin >> num;
		cin.ignore();
		
		getline(cin, cPos);
		
		istringstream iss(cPos);
		
		while(iss >> token)
		{
			vec.push_back(stoi(token));	
		}
		
		for(int i = 0; i < num; i++)
		{
		    pos = vec[i] - 1;
		    res += cLe[pos];
		}

        cout << res << endl;
        
        vec.clear();
	}
}
